app/frontend/components/domains/permit-application/checklist-sidebar.tsx

```diff
@@ -0,0 +1,81 @@
+import { Box, Divider, Flex, Heading, Tab, TabIndicator, TabList, Tabs, Text } from "@chakra-ui/react"
+import { CheckCircle, CircleDashed } from "@phosphor-icons/react"
+import { observer } from "mobx-react-lite"
+import React from "react"
+import { IPermitApplication } from "../../../models/permit-application"
+
+interface IChecklistSideBarProps {
+  permitApplication: IPermitApplication
+}
+
+export const ChecklistSideBar = observer(({ permitApplication }: IChecklistSideBarProps) => {
+  const { formJson } = permitApplication
+  const { selectedTabIndex, setSelectedTabIndex, indexOfBlockId, getBlockClass, getIsBlockPopulated } =
+    permitApplication
+
+  const handleTabsChange = (index: number, sectionId: string, blockId: string) => {
+    setSelectedTabIndex(index)
+    const className = getBlockClass(sectionId, blockId)
+    const element = document.getElementsByClassName(className)[0] as HTMLElement
```

Comment:
I believe you can use `document.querySelector` which will automatically return the first match. https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector